runner.py

```diff
@@ -110,68 +113,90 @@
             sys.exit(1)
         GlobalConfig(config_location=args.config_override)
 
-    runner = ScenarioRunner(name=args.name, uri=args.uri, uri_type=run_type, filename=args.filename,
-                    branch=args.branch, debug_mode=args.debug, allow_unsafe=args.allow_unsafe,
-                    skip_system_checks=args.skip_system_checks,
-                    skip_unsafe=args.skip_unsafe,verbose_provider_boot=args.verbose_provider_boot,
-                    full_docker_prune=args.full_docker_prune, dev_no_sleeps=args.dev_no_sleeps,
-                    dev_cache_build=args.dev_cache_build, dev_no_metrics=args.dev_no_metrics, dev_no_save=args.dev_no_save,
-                    dev_flow_timetravel=args.dev_flow_timetravel, dev_no_optimizations=args.dev_no_optimizations,
-                    docker_prune=args.docker_prune, dev_no_phase_stats=args.dev_no_phase_stats, user_id=args.user_id,
-                    skip_volume_inspect=args.skip_volume_inspect, commit_hash_folder=args.commit_hash_folder,
-                    usage_scenario_variables=variables_dict, phase_padding=not args.no_phase_padding)
+    # Use default filename if none provided
+    filename_patterns = args.filename if args.filename else ['usage_scenario.yml']
+    using_default_filename = not args.filename
+
+    filenames = []
+    for pattern in filename_patterns:
+        matches = glob.glob(pattern)
+        valid_files = [f for f in matches if os.path.isfile(f)]
+
+        if not valid_files:
+            if using_default_filename:
+                print(TerminalColors.FAIL, f'Error: Default file not found: {pattern}', TerminalColors.ENDC)
+                print('Please create the file or specify a different file with --filename')
+            else:
+                print(TerminalColors.FAIL, f'Error: No valid files found for --filename pattern: {pattern}', TerminalColors.ENDC)
+            sys.exit(1)
+        filenames.extend(valid_files)
+
+    # Remove duplicates and repeat for each iteration
+    filenames = list(set(filenames)) * args.iterations
```

Comment:
why remove duplicates? Maybe the user has a specifc repetition pattern in mind that shall be used? Vote to remove set()

Reply:
I have now removed the deduplication. This means that using the parameters `--filename usage_scenario.yml --filename usage_scenario.yml --iterations=2` would result in the file being executed four times.